### ISSUE TITLE
not loading correctly, but it does some times, grr

### ISSUE THREAD

**Issue by StuartClickReturn** [desandro/masonry#1196]
Hi there

My grid keeps loading with half the picture on top of one another, and cannot understand why?
https://newtopia1.wpenginepowered.com/artwork/

when you resize the window the grid reconfigures perfectly, why can it not just load right the first time... 

im confused where i have gone wrong
it looks like this --> https://newtopia1.wpenginepowered.com/wp-content/uploads/2024/03/Screenshot-2024-03-18-at-23.18.23.png or this --> https://newtopia1.wpenginepowered.com/wp-content/uploads/2024/03/Screenshot-2024-03-18-at-23.03.36.png

help?

**Comment by autoantwort** [comment]
I have the same problem. Have you found a solution? 

I guess the problem is that the layout is computed before the images are loaded and therefore the initial height that is not the final one is used.

**Comment by Luckyfella73** [comment]
Providing the JS you use for setting up the masonry would be good.

Do you use `imagesLoaded`? That is the recommended way to make sure the images are loaded before
doing the layout-math.

See documentation (scroll down to imagesLoaded section):
[https://masonry.desandro.com/layout](https://masonry.desandro.com/layout)

**Comment by autoantwort** [comment]
Yeah this was the problem/solution for me: https://masonry.desandro.com/layout#imagesloaded
